parfive/downloader.py

```diff
@@ -364,6 +377,10 @@ def callback(token, future, main_pb):
         """
         timeout = aiohttp.ClientTimeout(**timeouts)
         try:
+            kwargs['proxy'] = self.proxy
+            kwargs['proxy_auth'] = self.proxy_auth
+            kwargs['headers'] = self.ua_header
```

Comment:
You need to extract headers from the right place etc, and then do something like:
```python
if 'User-Agent' not in headers:
    headers['User-Agent'] = f"parfive {version}, aiohttp {version}, python {version}"
```